contracts/ronin/validator/CoinbaseExecution.sol

```diff
@@ -381,36 +399,37 @@ abstract contract CoinbaseExecution is
    * Emits the `BridgeOperatorSetUpdated` event.
    *
    */
-  function _revampRoles(uint256 _newPeriod, uint256 _nextEpoch, address[] memory currValidatorIds) private {
-    bool[] memory _maintainedList =
-      IMaintenance(getContract(ContractType.MAINTENANCE)).checkManyMaintainedById(currValidatorIds, block.number + 1);
+  function _removeInapplicableAndUpdateNewBlockProducers(
+    uint256 newPeriod,
+    uint256 nextEpoch,
+    address[] memory currValidatorIds
+  ) private {
+    uint256 nextBlock = block.number + 1;
+    bool[] memory maintainedList =
+      IMaintenance(getContract(ContractType.MAINTENANCE)).checkManyMaintainedById(currValidatorIds, nextBlock);
 
-    for (uint _i; _i < currValidatorIds.length;) {
-      address validatorId = currValidatorIds[_i];
+    // Remove block producer flag for all validators in the previous set
+    address[] memory prevBlockProducers = getBlockProducerIds();
+    uint256 length = prevBlockProducers.length;
+    for (uint256 i; i < length; ++i) {
+      address prevBlockProducerId = prevBlockProducers[i];
+
+      _validatorMap[prevBlockProducerId] =
+        _validatorMap[prevBlockProducerId].removeFlag(EnumFlags.ValidatorFlag.BlockProducer);
```

Comment:
try simplify this, by boolean, instead of enum flag
block producer = true/false.